### PR TITLE
Increased amount of prey cats can leap upon, including larva and facehuggers.

### PR DESCRIPTION
**Pull request by carlarctg**
# About the pull request

Increased amount of prey cats can leap upon, including larva and facehuggers. Others are bats, lizards, chicks, alien slugs, parrots.

Slightly updated cat hunting code. Very slightly. It was so bad man

# Explain why it's good for the game

It makes sense for cats to pounce on tiny critters. I also think this will lead to some very amusing hijinks.

It's just a little qol

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
add: Increased amount of prey cats can leap upon, including larva and facehuggers.
/:cl:

### CODE REVIEW

**Comment by stanalbatross** [comment]
This is add: not qol

**Comment by Ultimate-Fluff** [comment]
Pounce kills small critters, right? Oof, rip facehuggers.

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 1]
```diff
@@ -35,25 +46,29 @@
...
 					break
 
 	..()
 
-	for(var/mob/living/simple_animal/mouse/snack in oview(src, 3))
-		if(prob(15))
-			INVOKE_ASYNC(src, PROC_REF(emote), pick("hisses and spits!","mrowls fiercely!","eyes [snack] hungrily."))
+	for(var/mob/living/snack in oview(src, 3))
+		if(is_type_in_list(snack, hunting_targets) && prob(15) && snack.stat != DEAD)
+			visible_message(pick("\the hisses and spits!","\the mrowls fiercely!","\the eyes [snack] hungrily."))
```
Comment (carlarctg) [inline code/modules/mob/living/simple_animal/friendly/cat.dm:71]:
```suggestion
			visible_message(pick("\the [src] hisses at [snack]!", "\the [src] mrowls fiercely!", "\the [src] eyes [snack] hungrily."))
```

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 2]
```diff
@@ -35,25 +46,29 @@
...
 		if(!stat && !resting && !buckled)
-			for(var/mob/living/simple_animal/mouse/M in view(1,src))
-				if(!M.stat)
-					M.splat()
-					INVOKE_ASYNC(src, PROC_REF(emote), pick("bites \the [M]!","toys with \the [M].","chomps on \the [M]!"))
+			for(var/mob/living/prey in view(1,src))
+				if(is_type_in_list(prey, hunting_targets) && play_counter < 5)
+					prey.splat(src)
+					play_counter++
+					visible_message(pick("\the [src] bites \the [prey]!","\the [src] toys with \the [prey].","\the [src] chomps on \the [prey]!"))
```
Comment (harryob) [inline code/modules/mob/living/simple_animal/friendly/cat.dm:62]: \the is implicit here

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/living_helpers.dm [hunk 3]
```diff
@@ -17,3 +17,6 @@
 
 /mob/living/proc/check_fire_intensity_resistance()
 	return 0
+
+/mob/living/proc/splat(mob/killer)
```
Comment (Zonespace27) [inline code/modules/mob/living/living_helpers.dm:21]: code doc

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 4]
```diff
@@ -17,7 +17,18 @@
 	response_disarm = "gently pushes aside"
 	response_harm   = "kicks"
 	var/turns_since_scan = 0
-	var/mob/living/simple_animal/mouse/movement_target
+	var/mob/living/movement_target
```
Comment (Zonespace27) [inline code/modules/mob/living/simple_animal/friendly/cat.dm:20]: null on destroy; code doc all of these

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 5]
```diff
@@ -17,7 +17,18 @@
...
+	var/list/hunting_targets = list(
```
Comment (Zonespace27) [inline code/modules/mob/living/simple_animal/friendly/cat.dm:21]: static list

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 6]
```diff
@@ -17,7 +17,18 @@
...
+		/mob/living/simple_animal/mouse,
+		/mob/living/simple_animal/alien_slug,
+		/mob/living/simple_animal/bat,
+		/mob/living/simple_animal/parrot,
+		/mob/living/simple_animal/chick,
+		/mob/living/simple_animal/lizard,
+		/mob/living/carbon/xenomorph/facehugger,
+		/mob/living/carbon/xenomorph/larva
```
Comment (Zonespace27) [inline code/modules/mob/living/simple_animal/friendly/cat.dm:29]: trailing comma

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 7]
```diff
@@ -35,25 +46,29 @@
...
 		if(stat != DEAD)
 			if(++miaow_counter >= rand(20, 30)) //Increase the meow variable each tick. Play it at random intervals.
 				playsound(loc, "cat_meow", 15, 1, 4)
 				miaow_counter = 0 //Reset the counter
 		if(!stat && !resting && !buckled)
-			for(var/mob/living/simple_animal/mouse/M in view(1,src))
-				if(!M.stat)
-					M.splat()
-					INVOKE_ASYNC(src, PROC_REF(emote), pick("bites \the [M]!","toys with \the [M].","chomps on \the [M]!"))
+			for(var/mob/living/prey in view(1,src))
```
Comment (Zonespace27) [inline code/modules/mob/living/simple_animal/friendly/cat.dm:58]:
```suggestion
			for(var/mob/prey in view(1,src))
```
byond optimizes for the base types

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 8]
```diff
@@ -79,9 +94,12 @@
 		if( !movement_target || !(movement_target.loc in oview(src, 4)) )
 			movement_target = null
 			stop_automated_movement = 0
-			for(var/mob/living/simple_animal/mouse/snack in oview(src))
-				if(isturf(snack.loc) && !snack.stat)
+			for(var/mob/living/snack in oview(src))
```
Comment (Zonespace27) [inline code/modules/mob/living/simple_animal/friendly/cat.dm:97]:
```suggestion
			for(var/mob/snack in oview(src))
```

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 9]
```diff
@@ -35,27 +49,36 @@
 	if (PF)
 		PF.flags_pass = PASS_FLAGS_CRAWLER
 
+/mob/living/simple_animal/cat/Destroy()
+	QDEL_NULL(movement_target)
```
Comment (Zonespace27) [inline code/modules/mob/living/simple_animal/friendly/cat.dm:53]: don't QDEL_NULL; this'll delete whatever it's chasing if it gets deleted

**Review by Zonespace27 — APPROVED** [review]

**Pull request merged**